client/src/components/FriendRequest.tsx

```diff
@@ -0,0 +1,129 @@
+import { useEffect, useState } from 'react';
+import { FaUserFriends } from 'react-icons/fa';
+import { AiOutlineCheck } from 'react-icons/ai';
+import {
+  useGetUserFriendRequestsQuery,
+  useAcceptFriendRequestMutation,
+  useRejectFriendRequestMutation,
+} from '../redux/services/user/userService';
+import { User as UserType } from '../types';
+import { Spinner } from 'flowbite-react';
+import { Link } from 'react-router-dom';
+function FriendRequest() {
+  const { data, isLoading } = useGetUserFriendRequestsQuery('User');
+  const [friendRequests, setFriendRequests] = useState<UserType[]>();
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [acceptFriendRequest] = useAcceptFriendRequestMutation();
+  const [rejectFriendRequest] = useRejectFriendRequestMutation();
+  useEffect(() => {
+    if (data && !isLoading) {
+      setFriendRequests(data);
+    }
+  }, [data, isLoading]);
+
+  const handleBellClick = () => {
+    setIsDropdownOpen((prevState) => !prevState);
+  };
+
+  const handleIconClick = (event: any) => {
```

Comment:
If it's not too much of a hassle, try replacing the `any` type for the actual event type that corresponds to this case. Otherwise it's ok.